spark/src/test/scala/org/apache/spark/sql/CometTestBase.scala

```diff
@@ -231,11 +231,9 @@ abstract class CometTestBase
       df: => DataFrame): (Option[Throwable], Option[Throwable]) = {
     var expected: Option[Throwable] = None
     withSQLConf(CometConf.COMET_ENABLED.key -> "false") {
-      val dfSpark = Dataset.ofRows(spark, df.logicalPlan)
-      expected = Try(dfSpark.collect()).failed.toOption
+      expected = Try(Dataset.ofRows(spark, df.logicalPlan).collect()).failed.toOption
```

Comment:
if the plan parsing encounters a problem like `negative precision`, this will catch that.

Reply:
so the change here is just formatting: changing from 2 lines to 1?

Reply:
Earlier only df.collect() part was inside the Try, I also included df.logicalPlan - this includes any exception during parsing, in this example, it was throwing parsing error for this 
```sql
select a, cast(a as DECIMAL(10,-4)) from t order by a
```